### PR TITLE
Kubernetes secret config changes

### LINKED ISSUE #11461: Unable to read a literal value from a secret containing keys ending in "yaml,yml or properties"
**Describe the bug**
(Describe the problem clearly and concisely.)
When reading a secret any secret key with that ends with ".yml" or ".yaml" or ".properties" is assumed to contain a ConfigSource. It's not possible to read a literal value from a secret with containing keys ending in "yaml,yml or properties" 

**Expected behavior**
Should be able to read literal values from secrets containing keys ending in "yaml,yml or properties" 

**Actual behavior**
2020-08-19 09:42:27,386 DEBUG [io.qua.kub.cli.run.KubernetesConfigSourceProvider] (main) Attempting to read Secret tournamentcache-generated-secret
2020-08-19 09:42:48,593 DEBUG [io.qua.kub.cli.run.AbstractKubernetesConfigSourceUtil] (main) Attempting to convert data in Secret 'tournamentcache-generated-secret' to a list of ConfigSource objects
2020-08-19 09:45:23,074 DEBUG [io.qua.kub.cli.run.AbstractKubernetesConfigSourceUtil] (main) Secret 'tournamentcache-generated-secret' converted into 0ConfigSource objects
2020-08-19 09:46:09,200 DEBUG [io.qua.kub.cli.run.KubernetesConfigSourceProvider] (main) Done reading Secret tournamentcache-generated-secret

**To Reproduce**
1. 
2. 
3. 

**Configuration**
```properties
# Add your application.properties here, if applicable.

quarkus.kubernetes-config.enabled=true
quarkus.kubernetes-config.secrets.enabled=true
quarkus.kubernetes-config.fail-on-missing-config=true
quarkus.kubernetes-config.secrets=tournamentcache-generated-secret

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
```yaml
oc get secret -o yaml tournamentcache-generated-secret
apiVersion: v1
data:
  identities.yaml: Y3JlZGVudGlhbHM6Ci0gdXNlcm5hbWU6IGRldmVsb3BlcgogIHBhc3N3b3JkOiBUZDNaamZaYUJSOEQxQWZrCi0gdXNlcm5hbWU6IG9wZXJhdG9yCiAgcGFzc3dvcmQ6IGMyRm1KdTY3UlQ3R2JVQWsK
kind: Secret
metadata:
  creationTimestamp: "2020-08-09T08:29:49Z"
  labels:
    app: infinispan-secret-identities
    clusterName: tournamentcache
    infinispan_cr: tournamentcache
```

### PR DESCRIPTION
**Pull request by noelo**
Fixes https://github.com/quarkusio/quarkus/issues/11461

Added functionality to improve usage of kuberentes secrets as a ConfigSource

### CODE REVIEW

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ abstract ConfigSource createPropertiesConfigSource(String kubernetesConfigSource
             } else {
                 // TODO all keys named `*.{properties,yml,yaml}` are categorized as file sources,
                 //  but here, we only look for `application.{properties,yml,yaml}`
+                // Also see qualifier in categorize function below
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java:75]: The `TODO` is no longer valid, we should remove it, or replace with an up-to-date text.

File: extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java [hunk 2]
```diff
@@ -91,7 +92,8 @@ private static CategorizedConfigSourceData categorize(Map<String, String> data)
         List<Map.Entry<String, String>> fileData = new ArrayList<>();
         for (Map.Entry<String, String> entry : data.entrySet()) {
             String key = entry.getKey();
-            if (key.endsWith(".yml") || key.endsWith(".yaml") || key.endsWith(".properties")) {
+            if ((key.contains("application")) &&
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/AbstractKubernetesConfigSourceUtil.java:95]: I'd say `startsWith`, to be safe.

File: extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/SecretConfigSourceUtilTest.java [hunk 3]
```diff
@@ -102,14 +102,18 @@ void testWithAllKindsOfData() {
         List<ConfigSource> configSources = sut.toConfigSources(secret.getMetadata().getName(), secret.getData());
 
         assertThat(configSources).hasSize(4);
-        assertThat(configSources).filteredOn(c -> c.getName().toLowerCase().contains("literal"))
-                .hasOnlyOneElementSatisfying(c -> {
-                    assertThat(c.getProperties()).containsOnly(entry("some.key", "someValue"));
-                });
-        assertThat(configSources).filteredOn(c -> c.getName().toLowerCase().contains("application.properties"))
```
Comment (Ladicek) [inline extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/SecretConfigSourceUtilTest.java:109]: Why not keep the existing tests? They should still pass, no? At least the one that verifies content of `application.properties`.

**Comment by Ladicek** [comment]
Generally LGTM, but added a few minor comments.

File: extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/SecretConfigSourceUtilTest.java (hunk 3 above)
Comment (noelo) [reply extensions/kubernetes-config/runtime/src/test/java/io/quarkus/kubernetes/client/runtime/SecretConfigSourceUtilTest.java:109]: For some reason the existing code was adding a "-literal" to the secret key. No idea why, I searched the code and it wasn't removed so I removed it. It was on line 53

**Review by gsmet — DISMISSED** [review]
Just a quick note that this needs to be squashed before merging. Please dismiss once done.

**Review by Ladicek — APPROVED** [review]
LGTM from my side. Please squash per Guillaume's comment.

**Review by geoand — APPROVED** [review]
LGTM, thanks!

**Comment by geoand** [comment]
@gsmet not sure what the CI status is, as I don't want to merge and cause problems :)

**Comment by noelo** [comment]
Do I need to do anything to progress this PR ?

**Comment by geoand** [comment]
Can you please rebase it onto the latest master?

**Comment by noelo** [comment]
@geoand the CI tests keep failing due to test failures in other components. Do I just keep rebasing and hope the that CI eventually passes or do something else ?

**Comment by geoand** [comment]
I you have rebased onto the latest master and CI is still having problems, we most likely have some flaky tests we need to address.

I'll a look on Monday

**Comment by geoand** [comment]
Sorry, but now we have a git confict (although it's a very simple one), can you please resolve it and rebase onto master one more time? 

Thanks

**Comment by noelo** [comment]
good to go

**Comment by geoand** [comment]
Thanks a lot!

**Pull request merged**